ui-v2/tests/lib/measure/getMeasure.js

```diff
@@ -0,0 +1,26 @@
+/* eslint no-console: "off" */
+const log = function(results, measurement, tags) {
+  console.log(measurement, results, tags);
+};
+export default function(len = 10000, report = log, performance = window.performance) {
+  return function(cb, measurement, tags) {
+    let actual;
+    return new Array(len)
+      .fill(true)
+      .reduce(function(prev, item, i) {
+        return prev.then(function(ms) {
+          return new Promise(function(resolve) {
+            const start = performance.now();
+            cb().then(function(res) {
+              actual = res;
+              resolve(ms + (performance.now() - start));
+            });
+          });
+        });
+      }, Promise.resolve(0))
+      .then(function(total) {
+        report({ avg: total / len, total: total, count: len }, measurement, tags);
+        return actual;
+      });
+  };
+}
```

Comment:
Here's the measurement bits, this one measures things that run a promise based API - I have another version of this on the sidelines for non-promise based API's just not sure how to have them automatically switch in and out yet, or even if I want it to do it automatically.